bpf/init.sh

```diff
@@ -445,8 +447,16 @@ if [ "${TUNNEL_MODE}" != "<nil>" ]; then
 	if [ "$NODE_PORT" = "true" ]; then
 		COPTS="${COPTS} -DDISABLE_LOOPBACK_LB"
 	fi
-	bpf_load $ENCAP_DEV "$COPTS" "ingress" bpf_overlay.c bpf_overlay.o from-overlay ${CALLS_MAP}
-	bpf_load $ENCAP_DEV "$COPTS" "egress" bpf_overlay.c bpf_overlay.o to-overlay ${CALLS_MAP}
+
+	if ! bpf_load "$ENCAP_DEV" "$COPTS" ingress bpf_overlay.c bpf_overlay.o from-overlay "${CALLS_MAP}"; then
+		cilium bpf migrate-maps -e bpf_overlay.o -r 1 && exit 1
```

Comment:
Sorry, should have noticed this earlier, but couldn't we fold this into `bpf_load` (same for `bpf_load_cgroup`)? It would make most changes unnecessary and reduce code duplication.
```diff
- cilium bpf migrate-maps -e $OUT -r $RETCODE
+ if [ "$RETCODE" -ne 0 ]; then
+   cilium bpf migrate-maps -e bpf_overlay.o -r $RETCODE
+ fi
```

Reply:
Ah, yes. :) No need to even capture the return code if migration reverts are always done within those functions. Also removed the use of `set` in those functions. Thanks!